File: 0782-jewels-and-stones/0782-jewels-and-stones.swift
class Solution {
    func numJewelsInStones(_ jewels: String, _ stones: String) -> Int {
        var res:Int = 0
        var freq:[Character:Int] = [:]
        for stone in stones {
            freq[stone, default: 0] += 1
        }
        for jewel in jewels {
            if freq.keys.contains(jewel) {
                res = res + freq[jewel]!
            }
        }
        return res
    }
}

File: 0782-jewels-and-stones/0782-jewels-and-stones.dart
// Jewels and Stones
class Solution {
  int numJewelsInStones(String jewels, String stones) {
    Map<String, int> freq = {};
    int res = 0;
    for (int i = 0; i < stones.length; ++i) {
      freq[stones[i]] = (freq[stones[i]] ?? 0) + 1;
    }
    for (int i = 0; i < jewels.length; ++i) {
      if (freq.keys.contains(jewels[i])) {
        res = res + freq[jewels[i]]!;
      }
    }
    // print(freq);
    return res;
  }
}